src/runtime/callbacks.ts

```diff
@@ -0,0 +1,271 @@
+  triggerPayConfirmOpened(activityIframeView: ActivityIframeView): boolean {
+    return this.trigger_(CallbackId.PAY_CONFIRM_OPENED, activityIframeView);
+  }
+
+  setOnSubscribeRequest(callback: () => void): void {
+    this.setCallback_(CallbackId.SUBSCRIBE_REQUEST, callback);
+  }
+
+  /**
+   * @return Whether the callback has been found.
+   */
+  triggerSubscribeRequest(): boolean {
+    return this.trigger_(CallbackId.SUBSCRIBE_REQUEST, true);
+  }
+
+  hasSubscribeRequestCallback(): boolean {
+    return !!this.callbacks_[CallbackId.SUBSCRIBE_REQUEST];
+  }
+
+  setOnOffersFlowRequest(callback: () => void): void {
+    this.setCallback_(CallbackId.OFFERS_FLOW_REQUEST, callback);
+  }
+
+  /**
+   * @return Whether the callback has been found.
+   */
+  triggerOffersFlowRequest(): boolean {
+    return this.trigger_(CallbackId.OFFERS_FLOW_REQUEST, true);
+  }
+
+  hasOffersFlowRequestCallback(): boolean {
+    return !!this.callbacks_[CallbackId.OFFERS_FLOW_REQUEST];
+  }
+
+  setOnSubscribeResponse(
+    callback: (responsePromise: Promise<SubscribeResponse>) => void
+  ): void {
+    warn(
+      `[swg.js:setOnSubscribeResponse]: This method has been deprecated, please switch usages to 'setOnPaymentResponse'`
+    );
+    this.setCallback_(CallbackId.PAYMENT_RESPONSE, callback as Callback);
+  }
+
+  setOnContributionResponse(
+    callback: (responsePromise: Promise<SubscribeResponse>) => void
+  ): void {
+    warn(
+      `[swg.js:setOnContributionResponse]: This method has been deprecated, please switch usages to 'setOnPaymentResponse'`
+    );
+    this.setCallback_(CallbackId.PAYMENT_RESPONSE, callback as Callback);
+  }
+
+  setOnPaymentResponse(
+    callback: (responsePromise: Promise<SubscribeResponse>) => void
+  ): void {
+    this.setCallback_(CallbackId.PAYMENT_RESPONSE, callback as Callback);
+  }
+
+  /**
+   * @return Whether the callback has been found.
+   */
+  triggerPaymentResponse(responsePromise: Promise<SubscribeResponse>): boolean {
+    this.paymentResponsePromise_ = responsePromise.then(
+      (res) => {
+        this.trigger_(
+          CallbackId.PAYMENT_RESPONSE,
+          Promise.resolve(res.clone())
+        );
+      },
+      (reason) => {
+        if (isCancelError(reason)) {
+          return;
+        }
+        throw reason;
+      }
+    );
+    return !!this.callbacks_[CallbackId.PAYMENT_RESPONSE];
+  }
+
+  hasPaymentResponsePending(): boolean {
+    return !!this.resultBuffer_[CallbackId.PAYMENT_RESPONSE];
+  }
+
+  setOnFlowStarted(
+    callback: ({flow, data}: {flow: string; data: unknown}) => void
+  ): void {
+    this.setCallback_(CallbackId.FLOW_STARTED, callback as Callback);
+  }
+
+  /**
+   * @return Whether the callback has been found.
+   */
+  triggerFlowStarted(flow: string, data = {}): boolean {
+    return this.trigger_(CallbackId.FLOW_STARTED, {
+      flow,
+      data,
+    });
+  }
+
+  setOnFlowCanceled(
+    callback: (params: {flow: string; data: unknown}) => void
```

Comment:
`data` can be updated here, too, and similarly for `subscriptions.ts`

Reply:
Updated!